src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -3,88 +3,160 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections;
+using System.Collections.Generic;
 using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
 
 namespace System.Net
 {
     internal class HeaderInfoTable
     {
-        private static Hashtable HeaderHashTable;
-        private static HeaderInfo UnknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false);
+        private static readonly Func<string, string[]> s_singleParser = value => new[] { value };
+        private static readonly Func<string, string[]> s_multiParser = value => ParseValueHelper(value, isSetCookie: false);
+        private static readonly Func<string, string[]> s_setCookieParser = value => ParseValueHelper(value, isSetCookie: true);
+        private static readonly HeaderInfo s_unknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false, s_singleParser);
+        private static readonly Hashtable s_headerHashTable;
+        private static readonly Stack<char> s_valueParser = new Stack<char>();
+
+        private static string[] ParseValueHelper(string value, bool isSetCookie)
+        {
+            // RFC 6265: (for Set-Cookie header)
+            // If the name-value-pair string lacks a %x3D ("=") character, ignore the set-cookie-string entirely.
+            if (isSetCookie && (value.IndexOf('=') < 0)) return new string[0];
+
+            List<string> tempStringCollection = new List<string>();
+
+            bool inquote = false;
+            int chIndex = 0;
+            StringBuilder singleValue = new StringBuilder();
+            int insertPos = 0;
+
+            for (int i = 0; i < value.Length; i++)
+            {
+                if (value[i] == '\"')
+                {
+                    inquote = !inquote;
+                }
+                else if ((value[i] == ',') && !inquote)
+                {
+                    while (s_valueParser.Count > 0) singleValue.Insert(insertPos, s_valueParser.Pop());
+                    insertPos = singleValue.Length;
+                    if (!isSetCookie || !IsDuringExpiresAttributeParsing(singleValue.ToString()))
+                    {
+                        tempStringCollection.Add(singleValue.ToString().Trim());
+                        chIndex = 0;
+                        insertPos = 0;
+                        singleValue.Clear();
+                        continue;
+                    }
+                }
+                s_valueParser.Push(value[i]);
+                chIndex++;
+            }
+
+            // Now add the last of the header values to the stringtable.
+            if (chIndex != 0) {
+                while (s_valueParser.Count > 0) singleValue.Insert(insertPos, s_valueParser.Pop());
+                tempStringCollection.Add(singleValue.ToString().Trim());
+            }
+
+            s_valueParser.Clear();
+            string[] stringArray = tempStringCollection.ToArray();
+            return stringArray;
```

Comment:
You never use stringArray, so you might as well just return tempStringCollection.ToArray();

Reply:
Will fix.